subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/LeastRecentlyUsedCacheCleanup.java

```diff
@@ -33,26 +34,29 @@ public class LeastRecentlyUsedCacheCleanup extends AbstractCacheCleanup {
     private static final Logger LOGGER = LoggerFactory.getLogger(LeastRecentlyUsedCacheCleanup.class);
 
     public static final long DEFAULT_MAX_AGE_IN_DAYS_FOR_RECREATABLE_CACHE_ENTRIES = 7;
-    public static final long DEFAULT_MAX_AGE_IN_DAYS_FOR_EXTERNAL_CACHE_ENTRIES = 30;
 
     private final FileAccessTimeJournal journal;
-    private final long minimumTimestamp;
+    private final Supplier<Integer> numberOfDays;
 
-    public LeastRecentlyUsedCacheCleanup(FilesFinder eligibleFilesFinder, FileAccessTimeJournal journal, long numberOfDays) {
+    public LeastRecentlyUsedCacheCleanup(FilesFinder eligibleFilesFinder, FileAccessTimeJournal journal, Supplier<Integer> numberOfDays) {
         super(eligibleFilesFinder);
         this.journal = journal;
-        this.minimumTimestamp = Math.max(0, System.currentTimeMillis() - TimeUnit.DAYS.toMillis(numberOfDays));
+        this.numberOfDays = numberOfDays;
+    }
+
+    private long getMinimumTimestamp() {
+        return Math.max(0, System.currentTimeMillis() - TimeUnit.DAYS.toMillis(numberOfDays.get()));
     }
 
     @Override
     public void clean(CleanableStore cleanableStore, CleanupProgressMonitor progressMonitor) {
-        LOGGER.info("{} removing files not accessed on or after {}.", cleanableStore.getDisplayName(), new Date(minimumTimestamp));
+        LOGGER.info("{} removing files not accessed on or after {}.", cleanableStore.getDisplayName(), new Date(getMinimumTimestamp()));
```

Comment:
It might be useful to also print the number of days in this message, to give an understanding of what produces the minimum date the cleanup uses.